CenterForOpenScience/osf.io
# Feature/Add maintenance alerts to admin app [PLAT-170]

## Purpose

Adding maintenance alerts the the osf is more of a deal than it needs to be. 

## Changes

This adds setting maintenance alerts to the admin app so that they can be added at the click of a button.

Can quickly create or delete a maintenance banner from admin app. Does not allow maintenance banners to be updated - it's easy to replace a banner.

## QA Notes

The alert manager lives under 'Maintenance Alerts' in the admin app. Any time you add an alert it should appear on the OSF immediately (no need to check if the alert shows up on all pages - this is handled by the existing maintenance banner system).  If an alert is deleted it should disappear from the OSF immediately. 

Note: Maintenance alerts do not automatically go up/down based on the start and end times entered. They must be put up/down manually. These times are simply used in the default message ('The site will undergo maintenance between &lt;localized start time&gt;-&lt;localized end time&gt;. Thank you for your patience.') if no custom message is entered.

Another note: Times are entered in eastern time for convenience, but will show up in UTC when one is looking at the current alert in the admin app.

## Side Effects

None expected

## Ticket

https://openscience.atlassian.net/browse/PLAT-170

<img width="1219" alt="screen shot 2018-01-23 at 9 18 14 am" src="https://user-images.githubusercontent.com/7839433/35400294-593ac722-01c4-11e8-8008-7b31f25d6f70.png">
